### PR TITLE
v8: don't symlink private libraries

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

We shouldn't expose v8's private copies of `libicu*` and
`libchrome_zlib`. This links libraries that we consider keg-only on
macOS into `HOMEBREW_PREFIX`, and causes install conflicts on Linux.

Fixes install conflict spotted in #83593.

### CODE REVIEW

**Review by danielnachun — DISMISSED** [review]

**Comment by danielnachun** [comment]
I supposed we need to add an RPATH to the LDFLAGS?  I guess this was why I initially linked those, but I forgot about that.

**Comment by carlocab** [comment]
Might help, yea. Does `gn` respect `LDFLAGS`?

**Comment by danielnachun** [comment]
I thought it does, because I recall setting some `LDFLAGS` at one point.  This also reminds that this may have been why I tried to used the system `icu4c` instead of the bundled one.  The bundled `zlib` shouldn't be a problem because I think it actually has a different name so nothing else should link to it by accident.  Anyway I'm okay with either solution.

**Comment by cho-m** [comment]
Based on local install, `libexec` is not part of RPATH:
```console
# ldd /home/linuxbrew/.linuxbrew/Cellar/v8/9.2.230.22_1/lib/libv8.so
	libv8_libbase.so => /home/linuxbrew/.linuxbrew/lib/libv8_libbase.so (0x0000004003285000)
	libicui18n.so => /home/linuxbrew/.linuxbrew/lib/libicui18n.so (0x00000040032bd000)
	libicuuc.so => /home/linuxbrew/.linuxbrew/lib/libicuuc.so (0x000000400385d000)
	libchrome_zlib.so => not found
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x0000004003c75000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x0000004003e79000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x0000004004097000)
	libstdc++.so.6 => /home/linuxbrew/.linuxbrew/lib/gcc/11/libstdc++.so.6 (0x000000400429f000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x0000004004505000)
	libgcc_s.so.1 => /home/linuxbrew/.linuxbrew/lib/gcc/11/libgcc_s.so.1 (0x000000400480f000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x000000400482b000)
	/home/linuxbrew/.linuxbrew/lib/ld.so => /lib64/ld-linux-x86-64.so.2 (0x0000004000000000)
	libicudata.so.69 => /home/linuxbrew/.linuxbrew/Cellar/icu4c/69.1/lib/libicudata.so.69 (0x0000004004bf6000)

# objdump -p /home/linuxbrew/.linuxbrew/Cellar/v8/9.2.230.22_1/lib/libv8.so | grep RPATH
  RPATH                /home/linuxbrew/.linuxbrew/lib/gcc/11:/home/linuxbrew/.linuxbrew/lib:$ORIGIN:/home/linuxbrew/.linuxbrew/Cellar/v8/9.2.230.22_1/lib:/home/linuxbrew/.linuxbrew/opt/python@3.9/lib:/home/linuxbrew/.linuxbrew/opt/gmp/lib:/home/linuxbrew/.linuxbrew/opt/isl/lib:/home/linuxbrew/.linuxbrew/opt/mpfr/lib:/home/linuxbrew/.linuxbrew/opt/libmpc/lib:/home/linuxbrew/.linuxbrew/opt/zlib/lib:/home/linuxbrew/.linuxbrew/opt/zstd/lib:/home/linuxbrew/.linuxbrew/opt/binutils/lib:/home/linuxbrew/.linuxbrew/opt/gcc/lib:/home/linuxbrew/.linuxbrew/opt/ncurses/lib:/home/linuxbrew/.linuxbrew/opt/readline/lib:/home/linuxbrew/.linuxbrew/opt/libxml2/lib:/home/linuxbrew/.linuxbrew/opt/gettext/lib:/home/linuxbrew/.linuxbrew/opt/libffi/lib:/home/linuxbrew/.linuxbrew/opt/bzip2/lib:/home/linuxbrew/.linuxbrew/opt/pcre/lib:/home/linuxbrew/.linuxbrew/opt/gdbm/lib:/home/linuxbrew/.linuxbrew/opt/mpdecimal/lib:/home/linuxbrew/.linuxbrew/opt/openssl@1.1/lib:/home/linuxbrew/.linuxbrew/opt/sqlite/lib:/home/linuxbrew/.linuxbrew/opt/xz/lib:/home/linuxbrew/.linuxbrew/opt/expat/lib:/home/linuxbrew/.linuxbrew/opt/util-linux/lib:/home/linuxbrew/.linuxbrew/opt/glib/lib
```

**Comment by chrmoritz** [comment]
Maybe it's worth switching to system toolchain on Linux (not available on macOS), because then your LDFLAGS from the environment are respected: https://chromium.googlesource.com/chromium/src/build.git/+/refs/heads/main/toolchain/linux/unbundle/

File: Formula/v8.rb [hunk 1]
```diff
@@ -121,6 +122,8 @@ def install
```
Comment (chrmoritz) [inline Formula/v8.rb:122]: above that line to the on_linux block:

```ruby
gn_args[:custom_toolchain] = "\"//build/toolchain/linux/unbundle:default\""
gn_args[:host_toolchain] = "\"//build/toolchain/linux/unbundle:default\""
```

**Review by chrmoritz — CHANGES_REQUESTED** [review]

**Comment by carlocab** [comment]
> Maybe it's worth switching to system toolchain on Linux (not available on macOS), because then your LDFLAGS from the environment are respected: https://chromium.googlesource.com/chromium/src/build.git/+/refs/heads/main/toolchain/linux/unbundle/

Wait, if it's not using the system toolchain, what is it using then, and why do we have a dependency on GCC?

Am I just misunderstanding what you mean by switching to system toolchain here?

**Comment by chrmoritz** [comment]
System toolchain for V8 means basically, if respects what's passed in from the environment (`CC`, `CXX`, `LDFLAGS` and so on).

The current toolchain overrides these with whatever google things is right. You could even set `CC=gcc`and it would try to use `clang`. Switching to gcc is done currently by setting `gn_args[:is_clang] = false`.

Just compare these 2:
googles default gcc linux toolchain: https://chromium.googlesource.com/chromium/src/build.git/+/refs/heads/main/toolchain/linux/BUILD.gn#205
vs system toolchain from ENV: https://chromium.googlesource.com/chromium/src/build.git/+/refs/heads/main/toolchain/linux/unbundle/BUILD.gn#7

**Comment by chrmoritz** [comment]
I got it working in my tap! :tada:

Passed tap ci run: https://github.com/chrmoritz/homebrew-deno/runs/3508059158?check_suite_focus=true

One remaining question: Is it save to assume, that `nm` and `ar` are available on the path on Linux? Or how else I need to solve these 2 lines: https://github.com/chrmoritz/homebrew-deno/blob/e0b01425f4c3f1657e6ec452ba7aea17edc7d274/Formula/v8%409.2.230.22.rb#L119-L120

**Comment by chrmoritz** [comment]
should be fixed in #84650

**Pull request closed without merging**